extensions/rich_text_components/Image/directives/oppia-noninteractive-image.component.ts

```diff
@@ -58,13 +58,42 @@ export class NoninteractiveImage implements OnInit {
   constructor(
     private assetsBackendApiService: AssetsBackendApiService,
     private contextService: ContextService,
+    private changeDetectorRef: ChangeDetectorRef,
     private htmlEscaperService: HtmlEscaperService,
     private imageLocalStorageService: ImageLocalStorageService,
     private imagePreloaderService: ImagePreloaderService,
     private svgSanitizerService: SvgSanitizerService,
     private urlInterpolationService: UrlInterpolationService
   ) {}
 
+  private _loadBlobImage(url: string): void {
+    const img = new Image();
+    var c = document.createElement('canvas');
+    var ctx = c.getContext('2d');
+    img.onload = (e) => {
+      c.width = (e.target as HTMLImageElement).naturalWidth;
+      c.height = (e.target as HTMLImageElement).naturalHeight;
+      ctx.drawImage((e.target as HTMLImageElement), 0, 0);
+      c.toBlob((imgBlob) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(imgBlob);
+        reader.onloadend = () => {
+          if (imgBlob.type === 'image/svg+xml') {
+            this.imageUrl = (
+              this.svgSanitizerService.getTrustedSvgResourceUrl(
+                reader.result as string
+              )
+            );
+          } else {
+            this.imageUrl = reader.result;
+          }
+          this.changeDetectorRef.detectChanges();
+        };
+      }, 'image/png');
```

Comment:
What if the image is a GIF?

Please test the flow where a GIF is preloaded in the exploration player and rendered on the screen.

Reply:
Done!